src/ip/Address.cc

```diff
@@ -915,7 +915,10 @@ Ip::Address::fromHost(const char *host)
 
     char *tmp = xstrdup(start); // XXX: Slow. TODO: Bail on huge strings and use an on-stack buffer.
     tmp[strlen(tmp)-1] = '\0'; // XXX: Wasteful: xstrdup() just did strlen().
-    const bool result = lookupHostIP(tmp, true);
+    /* force result to false if IPv6 is disabled and the bracketed value is an IPv6 address */
```

Comment:
If this logic survives, please reflect this part of the API in the method description:
https://github.com/squid-cache/squid/blob/057a3cf34314d616325cffaca40a2cd49e486def/src/ip/Address.h#L230-L234

Reply:
Resolving this change request because the adjusted fromHost() implementation (if the method survives in this PR at all) is likely to match the existing description -- when EnableIpv6 is false, the method will return false for IPv6 addresses because the conversion will be unsuccessful.